arrow-fx-reactor/src/main/kotlin/arrow/fx/reactor/extensions/fluxk.kt

```diff
@@ -157,6 +176,160 @@ interface FluxKAsync :
+
+        sa.subscribe({
+          if (active.getAndSet(false)) {
+            emitter.next(RacePair.First(it, ffb))
+            emitter.complete()
+          }
+        }, { e -> onError(active, ddb, emitter, e) }, {})
+
+        sb.subscribe({
+          if (active.getAndSet(false)) {
+            emitter.next(RacePair.Second(ffa, it))
+            emitter.complete()
+          }
+        }, { e -> onError(active, dda, emitter, e) }, {})
+      }
+        .subscribeOn(scheduler)
+        .k()
+    }
+
+  fun <A, B> onError(
+    active: AtomicBooleanW,
+    other2: ReactorDisposable,
+    sink: FluxSink<RacePair<ForFluxK, A, B>>,
+    err: Throwable
+  ): Unit = if (active.getAndSet(false)) {
+    other2.dispose()
+    sink.error(err)
+  } else Unit
+
+  override fun <A, B, C> CoroutineContext.raceTriple(fa: FluxKOf<A>, fb: FluxKOf<B>, fc: FluxKOf<C>): FluxK<RaceTriple<ForFluxK, A, B, C>> =
+    asScheduler().let { scheduler ->
+      val active = AtomicBooleanW(true)
+      val forward = Disposables.composite()
+
+      Flux.create<RaceTriple<ForFluxK, A, B, C>> { emitter ->
+        val sa = ReplayProcessor.create<A>()
+        val sb = ReplayProcessor.create<B>()
+        val sc = ReplayProcessor.create<C>()
+
+        emitter.onCancel(forward)
+
+        val dda = fa.value()
+          .subscribeOn(scheduler)
+          .subscribe(sa::onNext, sa::onError)
+          .also { forward.add(it) }
+
+        val ddb = fb.value()
+          .subscribeOn(scheduler)
+          .subscribe(sb::onNext, sb::onError)
+          .also { forward.add(it) }
+
+        val ddc = fc.value()
+          .subscribeOn(scheduler)
+          .subscribe(sc::onNext, sc::onError)
+          .also { forward.add(it) }
+
+        val ffa = Fiber(sa.k(), FluxK { dda.dispose() })
+        val ffb = Fiber(sb.k(), FluxK { ddb.dispose() })
+        val ffc = Fiber(sc.k(), FluxK { ddc.dispose() })
+
+        sa.subscribe({
+          if (active.getAndSet(false)) {
+            emitter.next(RaceTriple.First(it, ffb, ffc))
+            emitter.complete()
+          }
+        }, { e -> onError(active, ddb, ddc, emitter, e) }, {})
+
+        sb.subscribe({
+          if (active.getAndSet(false)) {
+            emitter.next(RaceTriple.Second(ffa, it, ffc))
+            emitter.complete()
+          }
+        }, { e -> onError(active, dda, ddc, emitter, e) }, {})
+
+        sc.subscribe({
+          if (active.getAndSet(false)) {
+            emitter.next(RaceTriple.Third(ffa, ffb, it))
+            emitter.complete()
+          }
+        }, { e -> onError(active, dda, ddb, emitter, e) }, {})
+      }
+        .subscribeOn(scheduler)
+        .k()
+    }
+
+  private fun <A, B, C> onError(
+    active: AtomicBooleanW,
+    other2: ReactorDisposable,
+    other3: ReactorDisposable,
+    sink: FluxSink<RaceTriple<ForFluxK, A, B, C>>,
+    err: Throwable
+  ): Unit = if (active.getAndSet(false)) {
+    other2.dispose()
+    other3.dispose()
+    sink.error(err)
+  } else Unit
```

Comment:
both onError functions could be local, on the other hand we could merge them into one since we don't really care about the parametric types:

```suggestion
  private inline fun onError(
    active: AtomicBooleanW,
    other2: ReactorDisposable,
    other3: ReactorDisposable?,
    sink: FluxSink<*>,
    err: Throwable
  ): Unit = if (active.getAndSet(false)) {
    other2.dispose()
    other3?.dispose()
    sink.error(err)
  } else Unit
```

(feel free to replace the 2nd disposable nullable type with something fancy if you prefer, a vararg would also be possible as the last argument)